packages/itwinui-css/src/table/base.scss

```diff
@@ -320,6 +320,11 @@
   align-items: center;
   justify-content: center;
 
+  // If has at least one child
+  &:has(*) {
```

Comment:
Never use `:has(*)`, it's a very expensive selector. Firefox Devtools even displays a warning: https://bugzilla.mozilla.org/show_bug.cgi?id=1833679

You might be able to achieve the smae thing using `:not(:empty)`.